src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -3273,7 +3273,14 @@ def create_image_version(cmd, resource_group_name, gallery_name, gallery_image_n
         cred, _, _ = profile.get_login_credentials(resource=resource,
                                                    aux_subscriptions=aux_subscriptions)
         _, _, _, external_tokens = cred.get_all_tokens('https://management.azure.com/.default')
-        external_bearer_token = external_tokens[0][0] + ' ' + external_tokens[0][1]
+        if external_tokens:
+            external_token = external_tokens[0]
+            if len(external_token) >= 2:
+                external_bearer_token = external_token[0] + ' ' + external_tokens[1]
+            else:
+                logger.warning('Getting external tokens failed.')
+        else:
+            logger.warning('Getting external tokens failed.')
```

Comment:
is this a warning or error? It will cause command failed at later right?

Reply:
Finally, it will fail in service side if service requires external tokens.

Reply:
This `logger.warning()` goes to the stderr and breaks our Azure DevOps pipeline, which now thinks that `AzureCLI` task printed some error. The command itself (`az sig image-version create`) works fine as it did before, returns JSON result with   `"provisioningState": "Succeeded"`.